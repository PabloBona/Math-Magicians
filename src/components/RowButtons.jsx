import React from 'react';
import PropTypes from 'prop-types';

const RowButtons = ({ calculatorData, setCalculatorData, calculate }) => {
  const handleButtonClick = (buttonName) => {
    const newData = calculate(calculatorData, buttonName);
    setCalculatorData(newData);
  };

  return (
    <>
      <div className="col-8 col-6 d-flex justify-content-center flex-wrap w-50 bg">
        <button onClick={() => handleButtonClick('AC')} type="submit" className="btn btn-light rounded-0 border p-2 w-25 rounded-5 my-1 font-size ">AC</button>
        <button onClick={() => handleButtonClick('+/-')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">±</button>
        <button onClick={() => handleButtonClick('%')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">%</button>
        <button onClick={() => handleButtonClick('÷')} type="submit" className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1">÷</button>
        <button onClick={() => handleButtonClick('7')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">7</button>
        <button onClick={() => handleButtonClick('8')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">8</button>
        <button onClick={() => handleButtonClick('9')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">9</button>
        <button onClick={() => handleButtonClick('x')} type="submit" className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1">x</button>
        <button onClick={() => handleButtonClick('4')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">4</button>
        <button onClick={() => handleButtonClick('5')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">5</button>
        <button onClick={() => handleButtonClick('6')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">6</button>
        <button onClick={() => handleButtonClick('-')} type="submit" className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1">-</button>
        <button onClick={() => handleButtonClick('1')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">1</button>
        <button onClick={() => handleButtonClick('2')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">2</button>
        <button onClick={() => handleButtonClick('3')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">3</button>
        <button onClick={() => handleButtonClick('+')} type="submit" className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1">+</button>
        <button onClick={() => handleButtonClick('0')} type="submit" className="btn btn-light rounded-0 border py-2 w-50 rounded-5 my-1">0</button>
        <button onClick={() => handleButtonClick('.')} type="submit" className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1">.</button>
        <button onClick={() => handleButtonClick('=')} type="submit" className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1">=</button>
      </div>
    </>
  );
};

RowButtons.propTypes = {
  calculatorData: PropTypes.shape({
    total: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    next: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
    operation: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  }).isRequired,
  setCalculatorData: PropTypes.func.isRequired,
  calculate: PropTypes.func.isRequired,
};

export default RowButtons;
