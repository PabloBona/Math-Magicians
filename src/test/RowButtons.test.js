import React from 'react';
import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from '@testing-library/react';
import RowButtons from '../components/RowButtons';

describe('RowButtons component unit test', () => {
  const calculatorDataMock = {
    total: '0',
    next: null,
    operation: null,
  };
  const setCalculatorDataMock = jest.fn();
  const calculateMock = jest.fn();

  test('RowButtons Component Renders Properly', () => {
    const tree = renderer.create(
      <RowButtons
        calculatorData={calculatorDataMock}
        setCalculatorData={setCalculatorDataMock}
        calculate={calculateMock}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('RowButtons Component Calls handleButtonClick on Button Click', () => {
    render(
      <RowButtons
        calculatorData={calculatorDataMock}
        setCalculatorData={setCalculatorDataMock}
        calculate={calculateMock}
      />,
    );

    fireEvent.click(screen.getByText('7'));
    fireEvent.click(screen.getByText('6'));
    fireEvent.click(screen.getByText('5'));
    fireEvent.click(screen.getByText('4'));
    fireEvent.click(screen.getByText('x'));

    expect(calculateMock).toHaveBeenCalledTimes(5);
    expect(calculateMock).toHaveBeenCalledWith(calculatorDataMock, '7');
    expect(calculateMock).toHaveBeenCalledWith(calculatorDataMock, 'x');
  });
});
