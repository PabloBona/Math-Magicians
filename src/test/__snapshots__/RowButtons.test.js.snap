// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RowButtons component unit test RowButtons Component Renders Properly 1`] = `
<div
  className="col-8 col-6 d-flex justify-content-center flex-wrap w-50 bg"
>
  <button
    className="btn btn-light rounded-0 border p-2 w-25 rounded-5 my-1 font-size "
    onClick={[Function]}
    type="submit"
  >
    AC
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    ±
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    %
  </button>
  <button
    className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    ÷
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    7
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    8
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    9
  </button>
  <button
    className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    x
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    4
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    5
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    6
  </button>
  <button
    className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    -
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    1
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    2
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    3
  </button>
  <button
    className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    +
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-50 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    0
  </button>
  <button
    className="btn btn-light rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    .
  </button>
  <button
    className="btn btn-warning rounded-0 border py-2 w-25 rounded-5 my-1"
    onClick={[Function]}
    type="submit"
  >
    =
  </button>
</div>
`;
